### PR TITLE
Save order of resources in course section

### PR DESCRIPTION
**Pull request by OlyaKorchan**
Save order of resource in course section after add or drag and drop

https://github.com/ita-social-projects/SpaceToStudy-Client/assets/17857767/86c3e6e1-47b6-44a3-b576-2a7d5d65245b

### CODE REVIEW

**Review by dmtrth25 — APPROVED** [review]

**Review by Tolik170 — APPROVED** [review]

**Review by abalanovsky — APPROVED** [review]

**Review by TSlashDreamy — APPROVED** [review]

**Review by MariaAfonina — APPROVED** [review]

File: src/containers/course-section/CourseSectionContainer.tsx [hunk 1]
```diff
@@ -90,24 +95,42 @@ const CourseSectionContainer: FC<SectionProps> = ({
         ...sectionData.quizzes,
         ...sectionData.attachments
       ]
-      const updatedResourcesItems = prevResources.filter((prevResource) =>
-        allResourcesItems.some(
-          (currentResource) => currentResource._id === prevResource._id
+
```
Comment (maarekua) [inline src/containers/course-section/CourseSectionContainer.tsx:98]: setResources function is too complex. I suggest splitting it into a couple of functions.

File: src/containers/course-section/CourseSectionContainer.tsx [hunk 2]
```diff
@@ -90,24 +95,42 @@ const CourseSectionContainer: FC<SectionProps> = ({
...
-          updatedResourcesItems.push(currentResource)
+          updatedResourcesItems.push({
+            ...currentResource,
+            order: displayOrder.indexOf(currentResource._id)
+          })
         }
       }
       return updatedResourcesItems
     })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (maarekua) [inline src/containers/course-section/CourseSectionContainer.tsx:127]: I don't recommend skipping any of eslint rules.

File: src/pages/create-course/CreateCourse.tsx [hunk 3]
```diff
@@ -94,6 +94,17 @@ const CreateCourse = () => {
     handleNonInputValueChange('sections', value)
   }
 
+  const handleSectionResourcesOrder = (
+    id: string,
+    resources: CourseResources[]
+  ) => {
+    const sectionToEdit = data.sections.find((section) => section.id === id)
+    const orderedResources = resources.map((resource) => resource._id)
+    if (sectionToEdit) {
```
Comment (maarekua) [inline src/pages/create-course/CreateCourse.tsx:103]: You should move the resources map under the if statement. Cause there is no sense to map it if no item was found.

**Review by maarekua — CHANGES_REQUESTED** [review]

File: src/containers/course-section/CourseSectionContainer.tsx (hunk 1 above)
Comment (OlyaKorchan) [reply src/containers/course-section/CourseSectionContainer.tsx]: Refactored setResources and splitted it into three smaller functions

File: src/containers/course-section/CourseSectionContainer.tsx (hunk 2 above)
Comment (OlyaKorchan) [reply src/containers/course-section/CourseSectionContainer.tsx]: After refactoring of setResources there was no need for this line, so removed

File: src/pages/create-course/CreateCourse.tsx (hunk 3 above)
Comment (OlyaKorchan) [reply src/pages/create-course/CreateCourse.tsx:115]: moved

**Review by maarekua — APPROVED** [review]

**Pull request merged**